sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java

```diff
@@ -78,27 +78,33 @@ private HttpConnector(Builder builder, JavaVersion javaVersion) {
       // the Basic credentials consider an empty password.
       this.credentials = Credentials.basic(builder.login, nullToEmpty(builder.password));
     }
-
-    if (builder.proxy != null) {
-      this.okHttpClient.setProxy(builder.proxy);
-    }
     // proxy credentials can be used on system-wide proxies, so even if builder.proxy is null
     if (isNullOrEmpty(builder.proxyLogin)) {
       this.proxyCredentials = null;
     } else {
       this.proxyCredentials = Credentials.basic(builder.proxyLogin, nullToEmpty(builder.proxyPassword));
     }
+    this.okHttpClient = buildClient(builder, javaVersion);
+  }
+
+  private static OkHttpClient buildClient(Builder builder, JavaVersion javaVersion) {
+    OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
+    if (builder.proxy != null) {
+      okHttpClientBuilder.proxy(builder.proxy);
+    }
 
-    this.okHttpClient.setConnectTimeout(builder.connectTimeoutMs, TimeUnit.MILLISECONDS);
-    this.okHttpClient.setReadTimeout(builder.readTimeoutMs, TimeUnit.MILLISECONDS);
+    okHttpClientBuilder.connectTimeout(builder.connectTimeoutMs, TimeUnit.MILLISECONDS);
+    okHttpClientBuilder.readTimeout(builder.readTimeoutMs, TimeUnit.MILLISECONDS);
 
     ConnectionSpec tls = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
       .allEnabledTlsVersions()
       .allEnabledCipherSuites()
       .supportsTlsExtensions(true)
       .build();
-    this.okHttpClient.setConnectionSpecs(asList(tls, ConnectionSpec.CLEARTEXT));
-    this.okHttpClient.setSslSocketFactory(createSslSocketFactory(javaVersion));
+    okHttpClientBuilder.connectionSpecs(asList(tls, ConnectionSpec.CLEARTEXT));
+    okHttpClientBuilder.sslSocketFactory(createSslSocketFactory(javaVersion));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "sslSocketFactory"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)